src/main/scala/scalaz/stream/Process.scala

```diff
@@ -871,6 +871,31 @@ object Process extends ProcessInstances {
   def iterateEval[F[_], A](start: A)(f: A => F[A]): Process[F, A] =
     emit(start) ++ await(f(start))(iterateEval(_)(f))
 
+  /**
+   * Create a Process from an iterator. This should not be used directly,
+   * because iterators are mutable.
+   */
+  private [stream] def iteratorGo[O](iterator: Iterator[O]): Process[Task, O] = {
```

Comment:
Having this here and used by functions in `io` makes me feel squicky, especially since `io.iterator` and `Process.iterator` are so similar.  Not sure what to do about it though.  I was playing around with eliminating this function and just defining `Process.iterator` in terms of `io.iterator`, but I think it makes things ugly.